firebase-database/src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java

```diff
@@ -140,11 +139,12 @@ private ConnectionRequestCallback getOnComplete() {
     }
 
     private boolean markSent() {
-      return sent.compareAndSet(false, true);
-    }
-
-    private boolean wasSent() {
-      return sent.get();
+      boolean prev = sent;
+      if (prev) {
+        return false;
+      }
+      this.sent = true;
+      return true;
```

Comment:
Looks like we can inline `prev` and drop the `this.`.

This function should probably also have a short comment on what the return type is - or a more descriptive name.